ui/seek_bar.js

```diff
@@ -120,6 +163,97 @@ shaka.ui.SeekBar = class extends shaka.ui.RangeElement {
           this.onAdCuePointsChanged_();
         });
 
+    this.eventManager.listen(this.bar, 'mousemove', (event) => {
+      if (!this.player.getImageTracks().length) {
+        this.hideThumbnail_();
+        return;
+      }
+      const rect = this.bar.getBoundingClientRect();
+      const min = parseFloat(this.bar.min);
+      const max = parseFloat(this.bar.max);
+      // Pixels from the left of the range element
+      const mousePosition = event.clientX - rect.left;
+      // Pixels per unit value of the range element.
+      const scale = (max - min) / rect.width;
+      // Mouse position in units, which may be outside the allowed range.
+      const value = Math.round(min + scale * mousePosition);
+      // Show Thumbnail
+      this.showThumbnail_(mousePosition, value);
+    });
+
+    this.eventManager.listen(this.bar, 'keydown', (event) => {
+      this.hideThumbnailTimer_.stop();
+      const leftKeyCode = 37;
+      const rightKeyCode = 39;
+      // Left and Right only
+      if (event.keyCode != leftKeyCode && event.keyCode != rightKeyCode) {
+        return;
+      }
+      this.isMoving_ = true;
+      const min = parseFloat(this.bar.min);
+      const max = parseFloat(this.bar.max);
+      if (event.keyCode == leftKeyCode) {
+        this.bar.value = parseFloat(this.bar.value) - (max - min) * 0.025;
+      } else if (event.keyCode == rightKeyCode) {
+        this.bar.value = parseFloat(this.bar.value) + (max - min) * 0.025;
+      }
+      const rect = this.bar.getBoundingClientRect();
+      const value = Math.round(this.bar.value);
+      const scale = (max - min) / rect.width;
+      const position = (value - min) / scale;
+      this.showThumbnail_(position, value);
+      event.preventDefault();
+    });
+
+    this.eventManager.listen(this.bar, 'keyup', () => {
+      if (this.isMoving_) {
+        this.isMoving_ = false;
+        this.hideThumbnailTimer_.stop();
+        this.hideThumbnailTimer_.tickAfter(/* seconds= */ 0.25);
+      }
+    });
+
+    if (navigator.maxTouchPoints > 0) {
+      const touchMove = (event) => {
+        this.isMoving_ = true;
```

Comment:
Similarly, `shaka.ui.RangeElement`, the parent class for this class, has handlers for these touch events already, which calls `onChangeStart` and `onChange` and `onChange`. If you move the handling code to there, you shouldn't need these handlers.

Reply:
Done!